evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -60,27 +59,10 @@ def _prepare_data(self, X, y):
             pd.DataFrame, pd.Series: Prepared X and y data as pandas types
         """
         X = infer_feature_types(X)
-        int_nullable_cols = X.ww.select(IntegerNullable).columns
-        if len(int_nullable_cols) > 0:
-            try:
-                X = X.astype(
-                    {
-                        null_col: int
-                        for null_col in X.ww.select(IntegerNullable).columns
-                    },
-                )
-            except ValueError:
-                X = X.astype(
-                    {
-                        null_col: float
-                        for null_col in X.ww.select(IntegerNullable).columns
-                    },
-                )
-            X.ww.init(schema=X.ww.schema)
```

Comment:
This is nullable type handling that we can now remove.

Though worth noting that because the above `astype` calls are outside of woodwork, `X` had no ww types at this point, and this init call was actually with `schema=None` and we were just rerunning woodwork inference, which could've resulted in different oversamplers being chosen if inference changed which columns were Categorical (not super likely, which is why it never came up, I'm guessing)

So now we get to avoid the extra woodwork inference as well :)

Reply:
removed! fd06f5912